mitodl/ocw-to-hugo
# display updated course numbers

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-to-hugo/issues/256

#### What's this PR do?
In https://github.com/mitodl/ocw-data-parser/pull/140 the `new_course_numbers` field was added, which is a list of updates to either the primary course number (based on `department_number` and `master_course_number`) or any of the linked courses in `extra_course_number`.  This PR adds logic to the `getCourseNumbers` function in `helpers.js` that searches the `new_course_numbers` array for an update to a course number before adding it to the array that it returns.  If an update exists, a string is returned with the new course number followed by "formerly" and the old course number.

#### How should this be manually tested?
Convert a course with a course number update in the parsed json, such as `16-89j-space-systems-engineering-spring-2007` or `18-443-statistics-for-applications-spring-2015` and ensure that there are updated course numbers in the `data/course.json` file in the `course_numbers` array.

#### Any background context you want to provide?
While https://github.com/mitodl/ocw-data-parser/pull/140 has been merged and released, https://github.com/mitodl/open-discussions/pull/3398 has not (as of the time of writing) so the parsed JSON on S3 will not be updated.  If this is to be reviewed and tested prior to a merge of the above `open-discussions` PR and a full run of `backpopulate_ocw_data`, the reviewer will need to clone `ocw-data-parser` and parse JSON for these courses manually.
